src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,2292 @@
+	const char *tag = cvrf_item_type_get_text(item_type);
+	bool no_err;
+	while (xmlStrcmp(xmlTextReaderConstLocalName(reader), BAD_CAST tag) == 0) {
+		if (item_type == CVRF_REVISION) {
+			no_err = oscap_list_add(list, cvrf_revision_parse(reader));
+		} else if (item_type == CVRF_REFERENCE || item_type == CVRF_DOCUMENT_REFERENCE) {
+			no_err = oscap_list_add(list, cvrf_reference_parse(reader));
+		} else if (item_type == CVRF_ACKNOWLEDGMENT) {
+			no_err = oscap_list_add(list, cvrf_acknowledgment_parse(reader));
+		} else if (item_type == CVRF_GROUP) {
+			no_err = oscap_list_add(list, cvrf_group_parse(reader));
+		} else if (item_type == CVRF_INVOLVEMENT) {
+			no_err = oscap_list_add(list, cvrf_involvement_parse(reader));
+		} else if (item_type == CVRF_PRODUCT_STATUS) {
+			no_err = oscap_list_add(list, cvrf_product_status_parse(reader));
+		} else if (item_type ==  CVRF_REMEDIATION) {
+			no_err = oscap_list_add(list, cvrf_remediation_parse(reader));
+		} else if (item_type == CVRF_THREAT) {
+			no_err = oscap_list_add(list, cvrf_threat_parse(reader));
+		} else if (item_type == CVRF_SCORE_SET) {
+			no_err = oscap_list_add(list, cvrf_score_set_parse(reader));
+		} else if (item_type == CVRF_VULNERABILITY) {
+			no_err = oscap_list_add(list, cvrf_vulnerability_parse(reader));
+		}
+		xmlTextReaderNextNode(reader);
+
+		if (!no_err) {
+			cvrf_set_parsing_error(tag);
+			break;
+		}
+	}
+}
+
+static char *cvrf_parse_element(xmlTextReaderPtr reader, const char *tagname, bool next_elm) {
+	char *elm_value = NULL;
+	if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), BAD_CAST tagname)) {
+		elm_value = oscap_element_string_copy(reader);
+	}
+	if (next_elm)
+		xmlTextReaderNextElement(reader);
+
+	return elm_value;
+}
+
+struct cvrf_index *cvrf_index_parse_xml(struct oscap_source *index_source) {
+	__attribute__nonnull__(index_source);
+
+	struct cvrf_index *index = cvrf_index_new();
+	cvrf_index_set_index_file(index, oscap_source_readable_origin(index_source));
+	struct oscap_stringlist *file_list = oscap_stringlist_new();
+	struct cvrf_model *model;
+
+	FILE* file = fopen(oscap_source_get_filepath(index_source), "r");
+	char line[256];
+
+	while (fgets(line, sizeof(line), file)) {
+		printf("%s", line);
+		oscap_trim(line);
+		oscap_stringlist_add_string(file_list, line);
+	}
+	fclose(file);
+
+	struct oscap_string_iterator *iterator = oscap_stringlist_get_strings(file_list);
+	while (oscap_string_iterator_has_more(iterator)) {
+		const char *filename = oscap_string_iterator_next(iterator);
+		model = cvrf_model_import(oscap_source_new_from_file(filename));
+		if (model)
+			oscap_list_add(index->models, model);
+	}
+	oscap_string_iterator_free(iterator);
+	oscap_source_free(index_source);
+
+	return index;
+}
+
+struct cvrf_model *cvrf_model_parse(xmlTextReaderPtr reader) {
+	__attribute__nonnull__(reader);
+
+	if (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_CVRF_DOC) ||
+			!(xmlTextReaderNodeType(reader) == XML_READER_TYPE_ELEMENT))
```

Comment:
Fixed- changed to 
if (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_CVRF_DOC)  != 0 || xmlTextReaderNodeType(reader) != XML_READER_TYPE_ELEMENT)